### ISSUE TITLE
dev: Fan page does not appear when long-press (hold) assigned hardware button

### ISSUE THREAD

**Issue by Doughboy68** [Blackymas/NSPanel_HA_Blueprint#1007]
When a dimmable light is assigned to a hardware button, Holding (long press) correctly brings up the correct light page.

This is not the case when a fan, which has speed controls, is assigned to a hardware button - long press does nothing.

I have not tested covers since I have none to test.

**Comment by edwardtfn** [comment]
Can you open the fan page for this same fan when in the buttons pages?
Could you please share the value of attribute `supported_features` of this fan entity?

**Comment by Doughboy68** [comment]
Yes I can open the fan page with long press on buttons page.

preset_modes: null
percentage: 0
percentage_step: 33.333333333333336
preset_mode: null
friendly_name: Master Bedroom Fan
supported_features: 1

**Comment by edwardtfn** [comment]
Could you please take a look at HA logs for some error related to this?

**Comment by Doughboy68** [comment]
I looked at logs and did not see any error related to this.
Maybe since fan page is new, there is something different/overlooked how this page is handled in the code for the hardware buttons.
I tried on both my panels running Dev.

When long pressing a button icon assigned to fan entity:

nspanel05 NSPanel event changed to {"page": "fan", "component": "currentpage", "value": "pagechange", "entity": "fan.master_bedroom_fan"}

When long pressing button assigned to same fan entity:

nspanel05 Left Button turned off
8:44:23 PM - Now
nspanel05 Left Button turned on
8:44:18 PM - 5 seconds ago

For comparison, right button with dimmable rgb light assigned:

nspanel05 Right Button turned off
8:44:39 PM - 20 seconds ago
nspanel05 NSPanel event changed to {"page": "light", "component": "currentpage", "value": "pagechange", "entity": "light.computer_room_ceiling"}
8:44:38 PM - 20 seconds ago
nspanel05 current page changed to light
8:44:38 PM - 20 seconds ago
nspanel05 Right Button turned on
8:44:37 PM - 21 seconds ago
